gradle-failure-reports/src/main/java/com/palantir/gradle/failurereports/FailureReportsRootPlugin.java

```diff
@@ -84,12 +81,14 @@ private void collectOtherTaskFailures(Project project, TaskProvider<FinalizerTas
                                     projectTasks)
                             .filter(FailureReportsRootPlugin::isAllowedTask)
                             .filter(FailureReporterResources::executedAndFailed)
-                            .map(ThrowableFailureReporter::maybeGetFailureReport)
-                            .flatMap(Optional::stream))));
+                            .map(ThrowableFailureReporter::getFailureReport))));
         });
     }
 
     private static boolean isAllowedTask(Task task) {
-        return !(task instanceof JavaCompile) && !(task instanceof Checkstyle) && !(task instanceof FinalizerTask);
+        return !(task instanceof JavaCompile)
```

Comment:
nit: I'd be tempted to use de morgan here and change it to `!(... || ... || ...)` rather than `(!... && !... && !...)`. I think reducing the number of negations makes it a bit more readable, and it's more efficient (although this probably doesn't matter)